test/js_test.go

```diff
@@ -7587,6 +7587,8 @@ func TestJetStreamSubscribeContextCancel(t *testing.T) {
 }
 
 func TestJetStreamClusterStreamLeaderChangeClientErr(t *testing.T) {
+	t.Skip("The 2.9 server changed behavior making this test fail now")
```

Comment:
What is failing? And does that mean that the test is no longer relevant or that there is a breaking change? I don't remember changes in server that would explain that.

Reply:
Remember we were over zealous in responding from non-leaders in the server and changed that behavior to make some of the server tests pass consistently.

This test now does have timeouts for calls to StreamInfo.